curvefs/src/volume/block_group_loader.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Date: Friday Mar 04 22:59:44 CST 2022
+ * Author: wuhanqing
+ */
+
+#include "curvefs/src/volume/block_group_loader.h"
+
+#include <glog/logging.h>
+
+#include <utility>
+#include <vector>
+
+#include "absl/memory/memory.h"
+#include "curvefs/proto/common.pb.h"
+#include "curvefs/src/volume/allocator.h"
+#include "curvefs/src/volume/block_device_client.h"
+#include "curvefs/src/volume/block_group_updater.h"
+#include "curvefs/src/volume/utils.h"
+#include "src/common/bitmap.h"
+
+namespace curvefs {
+namespace volume {
+
+using ::curve::common::BITMAP_UNIT_SIZE;
+using ::curve::common::BitRange;
+
+BitmapRange BlockGroupBitmapLoader::CalcBitmapRange() const {
+    BitmapRange range;
+
+    range.length = blockGroupSize_ / blockSize_ / BITMAP_UNIT_SIZE;
+
+    switch (bitmapLocation_) {
+        case BitmapLocation::AtStart: {
```

Comment:
Use a comment to indicate where the start and end are on the volume

Reply:
added in common.proto